File: 338-counting-bits/338-counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        
        def count_bits(num):
            
            count = 0
            while(num):
                
                if(num&1):
                    count +=1
                
                num >>=1
            
            return (count)
        
        ans = []
        for num in range(n+1):
            bits = count_bits(num)
            ans.append(bits)
        
        return ans

File: 338-counting-bits/338-counting-bits.rs
impl Solution {
    pub fn count_bits(n: i32) -> Vec<i32> {
        
        let n = (n+1) as usize;
        let mut ans = vec![0;n];
        
        for idx in 1..n{
            ans[idx] = if idx%2 == 0{
                ans[idx/2]
            }
            else{
                ans[idx-1] +1
            }
        }
        ans
    }
}